### PR TITLE
Added support for passing custom AMI though the ClusterOptions property

### PR DESCRIPTION
**Pull request by cosminilie**
Please review this pull request as it should fix https://github.com/pulumi/pulumi-eks/issues/33. I'm not too much of a typescript guy so I've reused some of the patterns already used in the package. If you have any improvement  suggestions let me know.

### CODE REVIEW

File: nodejs/eks/cluster.ts [hunk 1]
```diff
@@ -105,6 +105,13 @@ export interface ClusterOptions {
      */
     instanceType?: pulumi.Input<aws.ec2.InstanceType>;
 
+    /**
+     * The AMI to use for worker nodes. Defaults to the value of Amazon EKS - Optimized AMI if no value is provided.
+	 * More information about teh AWS eks optimized ami is available at https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html.
```
Comment (pgavlin) [inline nodejs/eks/cluster.ts]: "teh" -> "the"

File: nodejs/eks/cluster.ts [hunk 2]
```diff
@@ -105,6 +105,13 @@ export interface ClusterOptions {
...
+	 * Use the information provided by AWS if you want to build your own AMI.
+     */
+    customAmiId?: pulumi.Input<string>;
```
Comment (pgavlin) [inline nodejs/eks/cluster.ts]: "custom" -> "node"

File: nodejs/eks/cluster.ts [hunk 3]
```diff
@@ -456,21 +463,29 @@ ${customUserData}
...
-            filters: [{
-                name: "name",
-                values: [ "amazon-eks-node-*" ],
-            }],
-            mostRecent: true,
-            owners: [ "602401143452" ], // Amazon
-        }, { parent: this });
+
+        let amiId: pulumi.Input<string> = args.customAmiId!;
+
```
Comment (pgavlin) [inline nodejs/eks/cluster.ts]: nit: extra line

File: nodejs/eks/cluster.ts (hunk 1 above)
Comment (cosminilie) [reply nodejs/eks/cluster.ts]: Fixed

File: nodejs/eks/cluster.ts (hunk 2 above)
Comment (cosminilie) [reply nodejs/eks/cluster.ts]: Fixed

File: nodejs/eks/cluster.ts (hunk 3 above)
Comment (cosminilie) [reply nodejs/eks/cluster.ts]: Fixed

**Comment by cosminilie** [comment]
@pgavlin ping

**Comment by pgavlin** [comment]
Thanks, @cosminilie !

**Pull request merged**